azurerm/resource_arm_application_gateway.go

```diff
@@ -1245,6 +1270,54 @@ func flattenApplicationGatewayBackendHTTPSettings(input *[]network.ApplicationGa
 	return results, nil
 }
 
+func expandApplicationGatewayConnectionDraining(d map[string]interface{}) *network.ApplicationGatewayConnectionDraining {
+	connectionsRaw := d["connection_draining"].([]interface{})
+
+	var connectionDraining *network.ApplicationGatewayConnectionDraining
+	if len(connectionsRaw) > 0 {
+		connectionRaw := connectionsRaw[0].(map[string]interface{})
+
+		enabled := connectionRaw["enabled"].(bool)
+		var drainTimeout int32
+
+		if enabled {
+			drainTimeout = int32(connectionRaw["drain_timeout_in_sec"].(int))
+		}
+
+		connectionDraining = &network.ApplicationGatewayConnectionDraining{
+			Enabled:           utils.Bool(enabled),
+			DrainTimeoutInSec: utils.Int32(drainTimeout),
+		}
+	}
+
+	return connectionDraining
+}
+
+func flattenApplicationGatewayConnectionDraining(input *network.ApplicationGatewayBackendHTTPSettingsPropertiesFormat) []interface{} {
+	results := make([]interface{}, 0)
+
+	if connectionDraining := input.ConnectionDraining; connectionDraining != nil {
+		enabled := false
+		if connectionDraining.Enabled != nil {
+			enabled = *connectionDraining.Enabled
+		}
+
+		var drainTimeout int32
+		if enabled && connectionDraining.DrainTimeoutInSec != nil {
```

Comment:
We should probably always read this back into state unless there is a good reason not to